Pong.java

```diff
@@ -116,25 +215,31 @@ else if (key_down && !key_up) {
 				movePlayer (player, player.position + SPEED);
 			}
 		}
-		// CPU HARD
+		// If paddle is controlled by a hard CPU, move the paddle to the optimal precomputed player destination.
 		else if (player.getType() == Player.CPU_HARD) {
 			movePlayer (player, player.destination);
 		}
-		// CPU EASY
+		// If the paddle is controlled by an easy CPU, move the paddle towards the ball.
 		else if (player.getType() == Player.CPU_EASY) {
 			movePlayer (player, ball_y);
 		}
 	}
-	
-	// Draw
+
+    /**
+     * Fired once every 20 ms. Calculates the new position of the ball and paddles and redraws them at that location.
+     * @param g Graphics object used to render the ball and paddles.
+     */
+    // Draw
 	public void paintComponent (Graphics g) {
 		super.paintComponent (g);
 		
-		// Prepara il campo di gioco
+		// If this is the first frame to be rendered.
 		if (new_game) {
+            //Place the ball in the center of the screen
```

Comment:
I think the problem is that you've been using spaces whereas the original code has tabs. Please convert to tabs.